src/search-helpers/bbl.js

```diff
@@ -20,17 +20,19 @@ const bbl = async (string) => {
      LIMIT 5
   `;
 
-  if (tenDigits) {
-    return carto.SQL(SQL).then(rows =>
-      rows.map((row) => {
-        row.label = toTitleCase(row.address);
-        row.type = 'lot';
-        delete row.address;
-        return row;
-      }));
-  }
+  if (!tenDigits) return []
 
-  return [];
+  try {
+    const rows = await carto.SQL(SQL)
+    return rows.map((row) => {
+      row.label = toTitleCase(row.address);
+      row.type = 'lot';
+      delete row.address;
+      return row;
+    });
+  } catch (error) {
+    throw new Error(`Failed to search bbl helper for string: ${string}. Failed with error: ${error.response?.statusText ? error.response?.statusText : "Internal server error"}`)
```

Comment:
This pattern happens often enough that the construction of the error message might be worth generalizing into a utility function. I have no strong feelings on the matter, though.

Reply:
Very much agree :/ but I was having trouble with that. The loop in `search.js` has access to the string identifier for each helper(`neighborhood`, `geosearch-v2`, etc), so ideally we could create the error dynamically there instead. However, I couldn't make sense of how to integrate a try/catch into the `for of` loop there. I'm open to suggestions if you know how that might work.

Reply:
The juice doesn't seem to be worth the squeeze right now